trunk/source/Library-ModelCheckerUtils/src/de/uni_freiburg/informatik/ultimate/lib/modelcheckerutils/cfg/ThreadInstanceAdder.java

```diff
@@ -220,27 +210,23 @@ private void addForkOtherThreadTransition(final IIcfgForkTransitionThreadCurrent
 
 		final IcfgEdgeFactory ef = icfg.getCfgSmtToolkit().getIcfgEdgeFactory();
 
-		final UnmodifiableTransFormula forkTransformula;
-		{
-			final UnmodifiableTransFormula parameterAssignment = fsa.constructInVarsAssignment(
-					icfg.getCfgSmtToolkit().getSymbolTable(), icfg.getCfgSmtToolkit().getManagedScript(),
-					icfg.getCfgSmtToolkit().getInParams().get(threadInstanceName));
-			final UnmodifiableTransFormula threadIdAssignment =
-					fsa.constructThreadIdAssignment(icfg.getCfgSmtToolkit().getSymbolTable(),
-							icfg.getCfgSmtToolkit().getManagedScript(), Arrays.asList(threadIdVars));
-			final Set<IProgramVar> localVarsWithoutInParams =
-					icfg.getCfgSmtToolkit().getSymbolTable().getLocals(threadInstanceName).stream()
-							.filter(x -> !icfg.getCfgSmtToolkit().getInParams().get(threadInstanceName).contains(x))
-							.collect(Collectors.toSet());
-			final UnmodifiableTransFormula havocOfLocalVars = TransFormulaUtils.constructHavoc(localVarsWithoutInParams,
-					icfg.getCfgSmtToolkit().getManagedScript());
-			final List<UnmodifiableTransFormula> conjuncts =
-					Arrays.asList(parameterAssignment, threadIdAssignment, havocOfLocalVars);
-			forkTransformula = TransFormulaUtils.sequentialComposition(mLogger, mServices,
-					icfg.getCfgSmtToolkit().getManagedScript(), false, false, false,
-					XnfConversionTechnique.BOTTOM_UP_WITH_LOCAL_SIMPLIFICATION, SimplificationTechnique.NONE,
-					conjuncts);
-		}
+		final UnmodifiableTransFormula parameterAssignment = fsa.constructInVarsAssignment(
+				icfg.getCfgSmtToolkit().getSymbolTable(), icfg.getCfgSmtToolkit().getManagedScript(),
+				icfg.getCfgSmtToolkit().getInParams().get(threadInstanceName));
+		final UnmodifiableTransFormula threadIdAssignment =
+				fsa.constructThreadIdAssignment(icfg.getCfgSmtToolkit().getSymbolTable(),
+						icfg.getCfgSmtToolkit().getManagedScript(), Arrays.asList(threadIdVars));
+		final Set<IProgramVar> localVarsWithoutInParams =
+				icfg.getCfgSmtToolkit().getSymbolTable().getLocals(threadInstanceName).stream()
+						.filter(x -> !icfg.getCfgSmtToolkit().getInParams().get(threadInstanceName).contains(x))
+						.collect(Collectors.toSet());
+		final UnmodifiableTransFormula havocOfLocalVars =
+				TransFormulaUtils.constructHavoc(localVarsWithoutInParams, icfg.getCfgSmtToolkit().getManagedScript());
+		final List<UnmodifiableTransFormula> conjuncts =
+				Arrays.asList(parameterAssignment, threadIdAssignment, havocOfLocalVars);
+		final UnmodifiableTransFormula forkTransformula = TransFormulaUtils.sequentialComposition(mLogger, mServices,
+				icfg.getCfgSmtToolkit().getManagedScript(), false, false, false,
+				XnfConversionTechnique.BOTTOM_UP_WITH_LOCAL_SIMPLIFICATION, SimplificationTechnique.NONE, conjuncts);

```

Comment:
Could this block be moved into a method?

Reply:
Done

Reply:
Oh sorry, I meant the whole construction of `forkTransformula` (that was previously in a nested block).

Reply:
Ah I see, changed that